datatypes/src/collections/feature_collection.rs

```diff
@@ -714,6 +714,45 @@ where
     }
 }
 
+#[derive(Debug)]
+pub struct FeatureCollectionRow<Geometry> {
+    pub geometry: Geometry,
+    pub time_interval: TimeInterval,
+    pub data: HashMap<String, FeatureDataValue>,
+}
+
+pub struct FeatureCollectionIterator<'a, T> {
+    pub geometries: T,
+    pub time_intervals: std::slice::Iter<'a, TimeInterval>,
+    pub data: HashMap<String, FeatureDataRef<'a>>,
+    pub row_num: usize,
+}
+
+impl<'a, T, Geometry> Iterator for FeatureCollectionIterator<'a, T>
+where
+    T: std::iter::Iterator<Item = Geometry>,
+    Geometry: crate::primitives::GeometryRef,
```

Comment:
FeatureCollectionRow already contains GeometryRefs instead of the Geometries itself, but it's very unclear. Currently, you can only really see this at that type constraint. I will rename the generic type.